Subject: [PATCH] Let useApproxAdvectionInAdMode act in Tapenade builds

Generated from MITgcm_c69m/mods_tapenade_hooks/ against the tree at
MITgcm_c69m/MITgcm (checkpoint69m) by check_against_tree.sh.
Apply from the tree's root with: git apply -p1 0003-let-useapproxadvectioninadmode-act-in-tapenade-builds.patch

--- a/pkg/generic_advdiff/gad_advection.F
+++ b/pkg/generic_advdiff/gad_advection.F
@@ -187,7 +187,7 @@
 C     make local copies to be tampered with if necessary
       advectionScheme = advectionSchArg
       vertAdvecScheme = vertAdvecSchArg
-#ifdef ALLOW_AUTODIFF_TAMC
+#ifdef ALLOW_AUTODIFF
       IF ( inAdMode .AND. useApproxAdvectionInAdMode ) THEN
 C     In AD-mode, we change non-linear, potentially unstable AD advection
 C     schemes to linear schemes with more stability. So far only DST3 with
@@ -199,7 +199,7 @@
      &      vertAdvecScheme = ENUM_DST3
 C     here is room for more advection schemes as this becomes necessary
       ENDIF
-#endif /* ALLOW_AUTODIFF_TAMC */
+#endif /* ALLOW_AUTODIFF */
 
 #ifdef ALLOW_DIAGNOSTICS
 C--   Set diagnostics flags and suffix for the current tracer
